File: Python/Problem003.py
import math

def primes(limit):
    prime_list = []

    yield 2
    yield 3
    num = 3

    while num + 2 < limit:
        num += 2

        if any( num % prime == 0 for prime in prime_list ): continue

        prime_list.append(num)
        yield num

def prime_list(limit):
    return list(primes(limit))

def highest_prime_factor(number):
    
    limit = math.floor(math.sqrt(number))

    divisor = 2

    while divisor < limit and number != divisor:
        if number % divisor == 0:
            number /= divisor
            continue
        divisor += 1 if divisor == 2 else 2
    
    return number

if __name__ == "__main__":
    print highest_prime_factor(600851475143)

File: JavaScript/Problem003.js
function isPrime2(num) {
    //    alert(num);
    if (isPrime.primeList.indexOf(num) != -1) return true;

    for (var i=2;i <= num/2;i++) {
	//alert('i = ' + i);
        if (isPrime(i)) {
	    if ( num % i == 0 ) {
		return false;
	    }
	}
    }
  
    isPrime.primeList.push(num); // cache results
    return true;
}
isPrime2.primeList = [2];


function lpf(num) {
    for (var i=2;i <= num/2;i++) {
        if (num % i == 0) {
	    if (isPrime(i)) {
		var largestPrime = i;
	    }
	}
    }
    if (!largestPrime) return num; //number is prime
    return largestPrime;
}

function allPrimes(limit) {

    var primes = [2];

    function isPrime(x) {
	for (var j in primes) {
	    if (x % primes[j] == 0) return false;
	}
	return true;
    }

    if (limit < 2) return null;
  
    for (var i=3; i <= limit; i += 2) {
	if (isPrime(i)) primes.push(i);
    }

    return primes;
}